src/transcoder/transcodedialog.cpp

```diff
@@ -205,6 +209,20 @@ void TranscodeDialog::JobComplete(const QString& input, const QString& output,
 
   UpdateStatusText();
   UpdateProgress();
+
+  bool overwrite_existing = ui_->remove_original->isChecked();
+  QFileInfo input_fileinfo(input);
+  QFileInfo output_fileinfo(output);
+  bool same_extension = input_fileinfo.suffix() == output_fileinfo.suffix();
+  bool same_path =
+      input_fileinfo.absolutePath() == output_fileinfo.absolutePath();
```

Comment:
Could most of this be moved into the if block below?

Reply:
It could and has been